client/src/App.js

```diff
@@ -29,6 +31,19 @@ class App extends PureComponent {
       },
     ],
   };
+  addLink = (title, url) => {
```

Comment:
gern funktionalität noch auslagern.
`linksCopy` wird nicht neu zugewiesen, weshalb man dafür `const` nehmen kann. Du musst auch nicht pushen, du könntest auch einfach sowas machen wie
```js
this.setState(state => ({ links: [...state.links, { id: this.state.idCounter, ... }], idCounter: state.idCounter + 1 }))
```

Reply:
Dann musst du das auch nicht auslagern, forget it :)